packages/autofocus/src/index.ts

```diff
@@ -0,0 +1,66 @@
+import { createEffect, onMount, JSX, Accessor } from "solid-js";
+
+declare module "solid-js" {
+  namespace JSX {
+    interface Directives {
+      autofocus: boolean;
+    }
+  }
+}
+
+/**
+ * Directive to autofocus the element on render. Uses the native `autofocus` attribute to decided whether to focus.
+ *
+ * @param element - Element to focus.
+ * @param autofocus - Should this directive be enabled. defaults to false.
+ *
+ * @example
+ * ```ts
+ * import { autofocus } from "@solid-primitives/autofocus";
+ *
+ * <button use:autofocus autofocus={true}>Autofocused</button>;
+ * ```
+ */
+export const autofocus = (element: HTMLInputElement, autofocus: () => boolean) => {
```

Comment:
The type of `element` param should probably be `Element`. (in the readme you used a button for the example so.. )
You can use the `Directive` type from utils for it: https://github.com/solidjs-community/solid-primitives/blob/main/packages/utils/src/types.ts#L19
```suggestion
export const autofocus = (element: Element, autofocus?: () => boolean) => {
```
Also if you make the `autofocus` parameter optional, it could be used nicely with `ref` as well:
```tsx
<button autofocus ref={el => {
  autofocus(el)
}}/>
```
But I guess that `createAutofocus` is for that

Reply:
Changed it to `HTMLElement`.
I can't use `Directive` as it accepts an `Element` and I require an `HTMLElement` to `.focus()`.